src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -462,8 +463,9 @@ mongoc_topology_new (const mongoc_uri_t *uri, bool single_threaded)
          MONGOC_TOPOLOGY_MIN_RESCAN_SRV_INTERVAL_MS;
 
       /* a mongodb+srv URI. try SRV lookup, if no error then also try TXT */
-      prefixed_service = bson_strdup_printf ("_mongodb._tcp.%s", service);
-      if (!topology->rr_resolver (prefixed_service,
+      prefixed_hostname = bson_strdup_printf (
+         "_%s._tcp.%s", mongoc_uri_get_srv_service_name (uri), srv_hostname);
```

Comment:
This is the more significant change, correct? I'm glancing past a lot of the diff pertaining to the renaming of "service" to "hostname" and wanted to confirm this is more significant.

Reply:
Correct. This is the actual implementation of the `srvServiceName` feature for initial DNS seedlist discovery. Similarly for SRV polling on line line 819. The rest of the "service" -> "hostname" edits are just syntactic changes for consistency.

Note: if the diffs are overwhelming, I recommend comparing the individual commit diffs for the [feature implementation](https://github.com/mongodb/mongo-c-driver/pull/916/commits/9ec9d8a16d35b62875a7680f00fa8d1313c1c5cf) and [renaming](https://github.com/mongodb/mongo-c-driver/pull/916/commits/b03ccd8163a7df911587b7a07177fd7786d046ad) separately, as I deliberately made sure to keep them in separate commits.